internal/auth/auth.go

```diff
@@ -205,49 +193,61 @@ func (a *Authenticator) Wait(ctx context.Context, state State) (Result, error) {
 	}
 }
 
-func (a *Authenticator) getDeviceCode(ctx context.Context) (State, error) {
-	data := url.Values{
-		"client_id": []string{a.ClientID},
-		"scope":     []string{strings.Join(requiredScopes, " ")},
-		"audience":  []string{a.Audience},
-	}
+// GetDeviceCode kicks-off the device authentication flow by requesting
+// a device code from Auth0. The returned state contains the
+// URI for the next step of the flow.
+func (a *Authenticator) GetDeviceCode(ctx context.Context, additionalScopes []string) (State, error) {
```

Comment:
The only functional change here is the addition of the `additionalScopes` argument, which enables the passing of additional scopes when requesting a grant.